Modules/@babylonjs/react-native/BabylonModule.ts

```diff
@@ -1,33 +1,72 @@
 import { NativeModules } from 'react-native';
-import { NativeEngine } from '@babylonjs/core';
+import { Engine, NativeEngine } from '@babylonjs/core';
 
-// This global object is part of Babylon Native.
+const disposedPropertyName = "BabylonReactNative_IsDisposed";
+
+declare const global: {
+    nativeCallSyncHook: any;
+};
+const isRemoteDebuggingEnabled = !global.nativeCallSyncHook;
+
+// This global object is owned by Babylon Native.
 declare const _native: {
     whenGraphicsReady: () => Promise<void>;
-    engineInstance: NativeEngine;
-}
+};
+
+// This JSI-based global object is owned by Babylon React Native.
+// This will likely be converted to a TurboModule when they are fully supported.
+declare const BabylonNative: {
+    readonly initializationPromise: Promise<void>;
+    setEngineInstance: (engine: NativeEngine | null) => void;
+    reset: () => void;
+};
 
-const NativeBabylonModule: {
-    initialize(): Promise<boolean>;
-    whenInitialized(): Promise<boolean>;
-    reset(): Promise<boolean>;
+// This legacy React Native module is created by Babylon React Native, and is only used to bootstrap the JSI object creation.
+// This will likely be removed when the BabylonNative global object is eventually converted to a TurboModule.
+const BabylonModule: {
+    initialize(): Promise<void>;
```

Comment:
>initialize [](start = 4, length = 10)

is there a reason we aren't moving this initialization call to BabylonNative now?

Reply:
Yes. At this point in the code, the `BabylonNative` object has not been instantiated and injected into the JS execution context. This method serves as a way to "bootstrap" the initialization process and results in the `BabylonNative` instance being instantiated and injected into the JS execution context. After this is done, all other JS<==>C++ communication can go through the `BabylonNative` object.